pants

```diff
@@ -45,12 +43,30 @@ function tempdir {
   mktemp -d "$1"/pants.XXXXXX
 }
 
+function get_pants_ini_config_value {
+  config_key="$1"
+  valid_delimiters="[:=]"
+  optional_space="[[:space:]]*"
+  prefix="^${config_key}${optional_space}${valid_delimiters}${optional_space}"
+  remove_prefix=" s#${prefix}##p"
+  sed -ne "/${prefix}/${remove_prefix}" pants.ini
```

Comment:
The way sed commands work in this case is that `/<pattern>/ s#<replace-pattern>#<replacement>#<flags>` says "for lines matching `<pattern>`, run the following `s` command -- replace `<replace-pattern>` in the line with `<replacement>`, printing or doing whatever according to `<flags>`". In this case I think the sed command itself shouldn't be split up, because the actual `s` command is what is doing things here and it makes it feel spooky otherwise. Something like:
```bash
  sed -ne "/${prefix}/ s#${prefix}##p" pants.ini
```
I think makes what's happening very clear ("when encountering prefix, remove it"). The use of `#` is something I like to do -- people tend to use `/` but that tends to clash a lot with things I want to replace, imo. The sed info pages are very nice but we hopefully shouldn't be doing anything complex enough to require them in a script we distribute like this.

Reply:
Agreed, this is much better. Thanks for the thorough explanation! Have not used sed much.